kerastuner/protos/kerastuner.proto

```diff
@@ -0,0 +1,88 @@
+/* Copyright 2019 The KerasTuner Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+syntax = "proto3";
+
+package kerastuner;
+
+
+enum Sampling {
+    NONE = 0;
+    LINEAR = 1;
+    LOG = 2;
+    REVERSE_LOG = 3;
+}
+
+message Float {
+    string name = 1;
+    double min_value = 2;
+    double max_value = 3;
+    double step = 4;
+    Sampling sampling = 5;
+    double default = 6;
+}
+
+message Int {
+    string name = 1;
+    sint64 min_value = 2;
+    sint64 max_value = 3;
+    sint64 step = 4;
+    Sampling sampling = 5;
+    sint64 default = 6;
+}
+
+message Choice {
+    string name = 1;
+
+    message FloatValues {
+        repeated double values = 1;
+    }
+    message IntValues {
+        repeated sint64 values = 1;
+    }
+    message StringValues {
+        repeated string values = 1;
+    }
+    oneof values {
+        FloatValues float_values = 2;
+        IntValues int_values = 3;
+        StringValues string_values = 4;
+    }
+
+    oneof default {
+        sint64 int_default = 5;
+        double float_default = 6;
+        string string_default = 7;
+    }
+
+    bool ordered = 8;
+}
+
+message Boolean {
+    string name = 1;
+    bool default = 2;
+}
+
+message HyperParameters {
```

Comment:
Here and elsewhere, google.protobuf.Value may be a better choice, rather than implicit types.